ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -790,7 +790,7 @@ create_attributes_from_template(struct obj_attrs **out, void *template,
 	if (rc)
 		goto out;
 
-	if (get_attribute(attrs, PKCS11_CKA_LOCAL, NULL, NULL) !=
+	if (get_attribute(temp, PKCS11_CKA_LOCAL, NULL, NULL) !=
```

Comment:
I looks it should be the same for `PKCS11_CKA_KEY_GEN_MECHANISM` at line 799, no? think

Reply:
Yes, you are right. As I was reading the spec, similar restriction would also apply to attributes like CKA_ALWAYS_SENSITIVE and CKA_NEVER_EXTRACTABLE. Basically for all attributes marked with restrictions 2,4,6 in Table 10.  I will add the checks and corresponding testcases for these.